### PR TITLE
Fix timetick block caused by dml task pop failed

### PR DESCRIPTION
**Pull request by bigsheeper**
/kind bug

issue: https://github.com/milvus-io/milvus/issues/23169

### CODE REVIEW

File: internal/proxy/task_scheduler.go [hunk 1]
```diff
@@ -227,15 +227,19 @@ func (queue *dmTaskQueue) Enqueue(t task) error {
...
-	pChannels, dmt, err := queue.preAddPChanStats(t)
+	dmt := t.(dmlTask)
+	pChannels, err := dmt.getChannels()
 	if err != nil {
+		log.Warn("getChannels failed when Enqueue", zap.Any("tID", t.ID()), zap.Error(err))
 		return err
 	}
+
 	//2. enqueue dml task
+	queue.statsLock.Lock()
```
Comment (xiaofan-luan) [inline internal/proxy/task_scheduler.go:241]: move the lock before pChannels, err := dmt.getChannels() to guard the channel cache``

Comment (bigsheeper) [reply internal/proxy/task_scheduler.go:241]: channel cache is only used at `getChannels`, and `getChannels` is only used when task enqueues and dequeues, which will never have a race.

**Comment by bigsheeper** [comment]
rerun ut

**Comment by bigsheeper** [comment]
rerun ut

**Pull request merged**